feat: Add support for 'bun' package manager

ryoppippi opened the pull request:
fixes: #1078
This commit introduces support for the 'bun' package manager in the
TypiaSetupWizard and updates the documentation accordingly. 

If you created a new feature, then create the unit test function, too.

```bash
# COMPILE
npm run build

# RE-WRITE TEST PROGRAMS IF REQUIRED
npm run test:template

# BUILD TEST PROGRAM
npm run build:test

# DO TEST
npm run test
```

Learn more about the [CONTRIBUTING](CONTRIBUTING.md)

ryoppippi committed:
feat: Add support for 'bun' package manager

This commit introduces support for the 'bun' package manager in the
TypiaSetupWizard and updates the documentation accordingly. Users can
now select 'bun' as a package manager during setup and follow the
provided instructions in the documentation for using 'bun' with Typia.

ryoppippi committed:
feat(PackageManager): support multiple package managers

This commit introduces support for multiple package managers including
npm, pnpm, yarn, and bun. The `manager` property in the `PackageManager`
class now uses a new `Manager` type. The install command and dev option
for each manager are now stored in `installCmdTable` and `devOptionTable`
respectively. The command execution in the `install` method has been
updated to use these tables.

ryoppippi committed:
feat(.gitignore): add bun.lockb to ignored files

This commit adds the bun.lockb file to the .gitignore. This file is
generated by the bun package manager and should not be tracked in the
repository.

ryoppippi committed:
feat(test/package.json): create script alias because `bun` cannot identify `build` means directory or script

ryoppippi committed:
feat(deploy): add support for bun command

A new function `detectComanndAvailable` has been added to check if a
command is available in the system. If the `bun` command is available,
additional tests are run using it. This provides more flexibility and
coverage in testing.

ryoppippi committed:
feat: add setup for bun in GitHub Actions

This commit introduces a new step in the GitHub Actions workflow
for setting up bun. The version of bun used is the latest. This
change is necessary for the build process.

ryoppippi committed:
feat: Add 'test-esm/**' to GitHub Actions workflow triggers

This commit includes 'test-esm/**' in the paths that trigger the GitHub
Actions workflows for both push and pull_request events. This ensures
that the workflows run whenever changes are made to the 'test-esm'
directory.

samchon reviewed (commented):
As I'm using Windows laptop, I need to investigate about the `which` command.

ryoppippi committed:
feat: replace child_process with dax-sh for command detection

This commit replaces the usage of child_process with dax-sh in the
'detectCommandAvailable' function. This change makes the function
asynchronous and simplifies the command detection logic. The 'main'
function is also updated to be asynchronous to accommodate this change.

Additionally, dax-sh is added as a dependency in package.json.

ryoppippi commented on deploy/index.ts:
```diff
@@ -93,6 +99,11 @@ const main = (): void => {
   test(version)("test-esm")(["npm run build", "npm start"]);
+  /* if bun is available, test with bun */
+  if (await detectComanndAvailable("bun")) {
+    test(version)("test")(["bun --bun run build_run", "bun --bun run start"]);
+    test(version)("test-esm")(["bun --bun run build", "bun --bun run start"]);
+  }
```
This is the command where bun executes the test

ryoppippi commented on .github/workflows/build.yml:
```diff
@@ -18,6 +20,9 @@ jobs:
       - uses: actions/setup-node@v4
         with:
           node-version: 20.x
+      - uses: oven-sh/setup-bun@v1
+        with:
+          bun-version: latest
```
This is whrere GHA installs Bun

ryoppippi committed:
Merge remote-tracking branch 'up/master' into feature/add-bun-support
